dev-guide/feature-gate.md

```diff
@@ -0,0 +1,280 @@
+---
+title: FeatureGates, round 2
+authors:
+- @deads2k
+reviewers:
+- team-leads # so they know how featuregates are changing.  The old way will work, but it will be unnecessarily hard.
+approvers:
+- @jspeed # api-approvers are picking up some code maintenance and we have to merge FeatureGate changes
+- staff-eng # so when asked by new team leads, they have some sense of how to do this.
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+- @jspeed
+creation-date: 2023-03-29
+last-updated: 2023-03-29
+---
+
+# FeatureGates, round 2
+
+## Summary
+
+This enhancement aims to reduce the effort required to add a feature gate to TechPreviewNoUpgrade and to promote
+that feature gate to Default.
+Feature gates in OpenShift are enabled and disabled in a particular FeatureSet, mixing and matching is not allowed.
+Prior to this enhancement, it is necessary to vendor openshift/api into every impacted repository.
+After this enhancement, it will only be necessary to vendor into cluster-config-operator.
+
+## Motivation
+
+Reduce the effort required to add and promote feature gates, to encourage and eventually require using
+feature gates to introduce new features into OCP.
```

Comment:
I like the idea of making them easier to use, but I don't support the end goal of requiring feature gates. We should have the conversation about whether we will do that separately from this enhancement.

Reply:
I can strike the sentence (this enhancement is about making it easier to use them), but I included it for transparency about my objective.  If it prevents you from agreeing to improve the current state of affairs I can remove it.

Reply:
Struck. It's distracting.  Let's talk in a year.